flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedClientHandler.java

```diff
@@ -274,4 +313,49 @@ private void decodeBufferOrEvent(RemoteInputChannel inputChannel, NettyMessage.B
 			bufferOrEvent.releaseBuffer();
 		}
 	}
+
+	private void writeAndFlushNextMessageIfPossible(Channel channel) {
```

Comment:
Please add some javadoc with a hint how all `inputChannelsWithCredit` will be handled, i.e. one is written immediately, following ones after successful writes.